### PR TITLE
feat: :sparkles: add types for node-random-name

### PR DESCRIPTION
**Pull request by Ankan002**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/node-random-name/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    seed?: string;
+    random?: () => number;
+    male?: boolean;
+    female?: boolean;
+}
+
+declare function random_name(args?: NodeRandomNameArgs): string;
+
+declare namespace random_name {
+    interface NodeRandomNameArgs {
```
Comment (jakebailey) [inline types/node-random-name/index.d.ts:19]: This shouldn't be duplicated; if you want to have it exported, you should remove the outer one and refer to it as `random_name.NodeRandomNameArgs` or something.

File: types/node-random-name/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        first?: boolean;
```
Comment (jakebailey) [inline types/node-random-name/index.d.ts:20]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/node-random-name/index.d.ts (hunk 1 above)
Comment (Ankan002) [reply types/node-random-name/index.d.ts:19]: We can implement that

File: types/node-random-name/index.d.ts (hunk 2 above)
Comment (Ankan002) [reply types/node-random-name/index.d.ts:20]: Yeah that might be the case

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node-random-name/index.d.ts
Consider running `npx dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 42531f4b908f5100bf6ea3fb740e103fe6567c24
</p>

File: types/node-random-name/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Type definitions for node-random-name 1.0
+// Project: https://github.com/cscott/node-random-name
+// Definitions by: Ankan Bhattacharya <https://github.com/Ankan002>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
+declare function random_name(args?: random_name.NodeRandomNameArgs): string;
```
Comment (jakebailey) [inline types/node-random-name/index.d.ts:7]: Looking upstream, this function is actually called `getName`; that may be preferable.

Comment (Ankan002) [reply types/node-random-name/index.d.ts:6]: @jakebailey Just updated the code with both the changes

Comment (jakebailey) [reply types/node-random-name/index.d.ts]: Sure, but not this thread's. But, reading their upstream readme, they use this name too, so I guess it's fine.

File: types/node-random-name/index.d.ts (hunk 2 above)
Comment (Ankan002) [reply types/node-random-name/index.d.ts]: @jakebailey required changes to implement the above mentioned thing is done

**Review by jakebailey — APPROVED** [review]

File: types/node-random-name/index.d.ts (hunk 3 above)
Comment (Ankan002) [reply types/node-random-name/index.d.ts]: @jakebailey Actually I already implemented the name change.

**Comment by Ankan002** [comment]
Ready to merge

**Pull request merged**